pkg/dockerregistry/server/auth.go

```diff
@@ -295,10 +320,19 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 	}
 
 	// In case of docker login, hits endpoint /v2
-	if len(accessRecords) == 0 {
-		if err := verifyOpenShiftUser(ctx, osClient); err != nil {
+	if len(bearerToken) > 0 {
+		if user, userid, err := verifyOpenShiftUser(ctx, osClient); err != nil {
 			return nil, ac.wrapErr(ctx, err)
+		} else {
```

Comment:
style: no else block here because you're returning, make the scope for the variables `a, b, c := verify(...); if err !=nil {`